controllers/apps/transformer_component_reconfigure.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright (C) 2022-2024 ApeCloud Co., Ltd
+
+This file is part of KubeBlocks project
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package apps
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appsv1 "github.com/apecloud/kubeblocks/apis/apps/v1"
+	"github.com/apecloud/kubeblocks/pkg/common"
+	"github.com/apecloud/kubeblocks/pkg/configuration/core"
+	"github.com/apecloud/kubeblocks/pkg/controller/component"
+	configctrl "github.com/apecloud/kubeblocks/pkg/controller/configuration"
+	"github.com/apecloud/kubeblocks/pkg/controller/graph"
+	"github.com/apecloud/kubeblocks/pkg/controller/model"
+	intctrlutil "github.com/apecloud/kubeblocks/pkg/controllerutil"
+)
+
+// componentReloadActionSidecarTransformer handles component configuration render
+type componentReloadActionSidecarTransformer struct {
+	client.Client
+}
+
+var _ graph.Transformer = &componentReloadActionSidecarTransformer{}
+
+func (t *componentReloadActionSidecarTransformer) Transform(ctx graph.TransformContext, dag *graph.DAG) error {
+	transCtx, _ := ctx.(*componentTransformContext)
+
+	comp := transCtx.Component
+	cluster := transCtx.Cluster
+	compOrig := transCtx.ComponentOrig
+	synthesizeComp := transCtx.SynthesizeComponent
+
+	if model.IsObjectDeleting(compOrig) {
+		return nil
+	}
+	if common.IsCompactMode(compOrig.Annotations) {
+		transCtx.V(1).Info("Component is in compact mode, no need to create configuration related objects",
+			"component", client.ObjectKeyFromObject(transCtx.ComponentOrig))
+		return nil
+	}
+
+	reconcileCtx := &configctrl.ResourceCtx{
+		Context:       transCtx.Context,
+		Client:        t.Client,
+		Namespace:     comp.GetNamespace(),
+		ClusterName:   synthesizeComp.ClusterName,
+		ComponentName: synthesizeComp.Name,
+	}
+
+	var configmaps []*corev1.ConfigMap
+	cachedObjs := resolveRerenderDependOnObjects(dag)
+	for _, tpls := range [][]appsv1.ComponentTemplateSpec{synthesizeComp.ScriptTemplates, synthesizeComp.ConfigTemplates} {
+		objects, err := configctrl.RenderTemplate(reconcileCtx, cluster, synthesizeComp, comp, cachedObjs, tpls)
```

Comment:
Cloud you move the rendering out of the configuration and define it in a separate package, so that both apps and configuration can use it independently?

Reply:
And, when rendering config/script templates, simply replace the variables defined by the API, no need to provide unexposed objects and built-in functions.